{"ast":null,"code":"\"use strict\";\n\nvar __extends = this && this.__extends || function () {\n  var extendStatics = function (d, b) {\n    extendStatics = Object.setPrototypeOf || {\n      __proto__: []\n    } instanceof Array && function (d, b) {\n      d.__proto__ = b;\n    } || function (d, b) {\n      for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p];\n    };\n    return extendStatics(d, b);\n  };\n  return function (d, b) {\n    if (typeof b !== \"function\" && b !== null) throw new TypeError(\"Class extends value \" + String(b) + \" is not a constructor or null\");\n    extendStatics(d, b);\n    function __() {\n      this.constructor = d;\n    }\n    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());\n  };\n}();\nObject.defineProperty(exports, \"__esModule\", {\n  value: true\n});\nexports.PineconeRequestError = exports.PineconeConnectionError = void 0;\nvar base_1 = require(\"./base\");\n/**\n * This error is thrown when the client attempts to make a\n * request and never receives any response.\n *\n * This could be due to:\n * - Network problems which prevent the request from being completed.\n * - An outage of Pinecone's APIs. See [Pinecone's status page](https://status.pinecone.io/) to find out whether there is an ongoing incident.\n *\n * The `cause` property will contain a reference to the underlying error. Inspect its value to find out more about the root cause of the error.\n * ```\n * import { Pinecone } from '@pinecone-database/pinecone';\n *\n * const p = new Pinecone({ apiKey: 'invalid-api-key-value' })\n *\n * try {\n *  await p.listIndexes();\n * } catch (e) {\n *  console.log(e.name); // PineconeConnectionError\n *  console.log(e.cause); // Error [FetchError]: The request failed and the interceptors did not return an alternative response\n *  console.log(e.cause.cause); // TypeError: fetch failed\n *  console.log(e.cause.cause.cause); // Error: getaddrinfo ENOTFOUND controller.wrong-environment.pinecone.io\n * }\n * ```\n *\n * @see [Pinecone's status page](https://status.pinecone.io/)\n * */\nvar PineconeConnectionError = /** @class */function (_super) {\n  __extends(PineconeConnectionError, _super);\n  function PineconeConnectionError(e, url) {\n    var _this = this;\n    var urlMessage = '';\n    if (url) {\n      urlMessage = \" while calling \".concat(url);\n    }\n    _this = _super.call(this, \"Request failed to reach Pinecone\".concat(urlMessage, \". This can occur for reasons such as network problems that prevent the request from being completed, or a Pinecone API outage. Check your network connection, and visit https://status.pinecone.io/ to see whether any outages are ongoing.\"), e) || this;\n    _this.name = 'PineconeConnectionError';\n    return _this;\n  }\n  return PineconeConnectionError;\n}(base_1.BasePineconeError);\nexports.PineconeConnectionError = PineconeConnectionError;\n/**\n * This error is thrown any time a request to the Pinecone API fails.\n *\n * The `cause` property will contain a reference to the underlying error. Inspect its value to find out more about the root cause.\n */\nvar PineconeRequestError = /** @class */function (_super) {\n  __extends(PineconeRequestError, _super);\n  function PineconeRequestError(context) {\n    var _this = this;\n    if (context.response) {\n      _this = _super.call(this, \"Request failed during a call to \".concat(context.init.method, \" \").concat(context.url, \" with status \").concat(context.response.status), context.error) || this;\n    } else {\n      _this = _super.call(this, \"Request failed during a call to \".concat(context.init.method, \" \").concat(context.url), context.error) || this;\n    }\n    return _this;\n  }\n  return PineconeRequestError;\n}(base_1.BasePineconeError);\nexports.PineconeRequestError = PineconeRequestError;","map":{"version":3,"names":["base_1","require","PineconeConnectionError","_super","__extends","e","url","_this","urlMessage","concat","call","name","BasePineconeError","exports","PineconeRequestError","context","response","init","method","status","error"],"sources":["../../src/errors/request.ts"],"sourcesContent":[null],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,IAAAA,MAAA,GAAAC,OAAA;AAGA;;;;;;;;;;;;;;;;;;;;;;;;;;AA0BA,IAAAC,uBAAA,0BAAAC,MAAA;EAA6CC,SAAA,CAAAF,uBAAA,EAAAC,MAAA;EAC3C,SAAAD,wBAAYG,CAAQ,EAAEC,GAAY;IAAlC,IAAAC,KAAA;IACE,IAAIC,UAAU,GAAG,EAAE;IACnB,IAAIF,GAAG,EAAE;MACPE,UAAU,GAAG,kBAAAC,MAAA,CAAkBH,GAAG,CAAE;;YAGtCH,MAAA,CAAAO,IAAA,OACE,mCAAAD,MAAA,CAAmCD,UAAU,gPAA6O,EAC1RH,CAAC,CACF;IACDE,KAAI,CAACI,IAAI,GAAG,yBAAyB;;EACvC;EACF,OAAAT,uBAAC;AAAD,CAAC,CAb4CF,MAAA,CAAAY,iBAAiB;AAAjDC,OAAA,CAAAX,uBAAA,GAAAA,uBAAA;AAeb;;;;;AAKA,IAAAY,oBAAA,0BAAAX,MAAA;EAA0CC,SAAA,CAAAU,oBAAA,EAAAX,MAAA;EACxC,SAAAW,qBAAYC,OAAqB;IAAjC,IAAAR,KAAA;IACE,IAAIQ,OAAO,CAACC,QAAQ,EAAE;MACpBT,KAAA,GAAAJ,MAAA,CAAAO,IAAA,OACE,mCAAAD,MAAA,CAAmCM,OAAO,CAACE,IAAI,CAACC,MAAM,OAAAT,MAAA,CAAIM,OAAO,CAACT,GAAG,mBAAAG,MAAA,CAAgBM,OAAO,CAACC,QAAQ,CAACG,MAAM,CAAE,EAC9GJ,OAAO,CAACK,KAAc,CACvB;KACF,MAAM;MACLb,KAAA,GAAAJ,MAAA,CAAAO,IAAA,OACE,mCAAAD,MAAA,CAAmCM,OAAO,CAACE,IAAI,CAACC,MAAM,OAAAT,MAAA,CAAIM,OAAO,CAACT,GAAG,CAAE,EACvES,OAAO,CAACK,KAAc,CACvB;;;EAEL;EACF,OAAAN,oBAAC;AAAD,CAAC,CAdyCd,MAAA,CAAAY,iBAAiB;AAA9CC,OAAA,CAAAC,oBAAA,GAAAA,oBAAA","ignoreList":[]},"metadata":{},"sourceType":"script","externalDependencies":[]}